New page for release notes (#169)
EVerest/EVerest: docs/general/release-notes.rst

.. release_notes:

.. _release_notes_main:

#####################
EVerest Release Notes
#####################

See also our detailed release information based on all PRs and source code
commits on GitHub:
`EVerest releases <https://github.com/EVerest/everest-core/releases>`_

> The EVerest releases will come monthly. A release for a dedicated month can be

Reviewer: ```suggestion
EVerest releases are scheduled monthly. A release for a dedicated month can be
```

shall be related to, a charging rate unit and a charging schedule period.

For details, see
`the corresponding PR <https://github.com/EVerest/everest-core/pull/582>`_.

Note that this is an API-breaking change.

Rust now feature-complete
-------------------------

The feature-completeness of Rust in EVerest is meant in relation to the C++
implementation. Some recent features prepared the way for that status:

Having multislot support and a proper integration of Rust testing now in
EVerest, the integration of Rust can be seen as thorough as C++.

Payment terminal integration
----------------------------

Regarding payment integration, there is now added support for ZVT protocol
based devices. In EVerest, you can have "Eichrecht"-compliant payment on
charging points on board.

See a great overview of bank card payments here:
`Integration of bank card payment in EVerest <https://everest.github.io/nightly/general/06_handling_bank_cards.html>`_
.

Admin Panel improvements
------------------------

The EVerest Admin Panel has been part of EVerest for quite some time now. It
was time to improve the handling and stability of that.

> Please, be aware that it is still a beta-stage frontend tool which can be used

Reviewer: ```suggestion
Please be aware that it is still a beta-stage frontend tool which can be used
```

`Bazel in EVerest documentation <https://github.com/EVerest/EVerest/pull/162>`_
.

JsEvManager as replacement for JsCarSimulator
---------------------------------------------

The new module that replaces the JsCarSimulator is still a JavaScript based EV
simulator, but can be run on "real" hardware. This JsEvManager is the
counterpart of the EvseManager to be able to simulate charging sessions.

A C++ implementation will be coming soon.

2024.3.0 - LATEST STABLE
========================

Plug & Charge
-------------

The full process around a Plug&Charge session has been implemented, which
involves the communication to an electric vehicle and to systems in the cloud.

This means that the implementation has been done in ISO 15118 and OCPP code
parts of EVerest. For an overview and configuration instructions, see
`Plug and Charge Configuration in EVerest <https://everest.github.io/nightly/general/07_configure_plug_and_charge.html>`_
.

Fix for YetiDriver
------------------

The YetiDriver has been fully ported to the new BSP interface. For additional
information and changes, see
`the corresponding PR <https://github.com/EVerest/everest-core/pull/595>`_.

> MQTT topic for OCPP transaction state and ID

Reviewer: ```suggestion
Extended OCPP interface for transaction state and ID
```

--------------------------------------------

> The OCPP-related information of TransactionEvents are published as topics via
> MQTT now. Also the transaction ID coming from a CSMS is a newly published
> topic.

Reviewer: ```suggestion
The OCPP-related information of TransactionEvents are published as part of the [OCPP interface](https://everest.github.io/nightly/_generated/interfaces/ocpp.html). Also the transaction ID received from a CSMS is a now published.
```
